fe/fe-core/src/main/java/com/starrocks/catalog/FunctionSet.java

```diff
@@ -172,7 +172,7 @@
     private final Map<String, List<Function>> vectorizedFunctions;
     // This does not contain any user defined functions. All UDFs handle null values by themselves.
     private final ImmutableSet<String> nonNullResultWithNullParamFunctions = ImmutableSet.of("if", "hll_hash",
```

Comment:
should merge this set to `alwaysReturnNonNullableFunctions`

Reply:
The function `"if", "concat_ws", "ifnull", "nullif", "null_or_empty", "coalesce"` don't always return non-nullable result, while `hll_hash`, `bitmap_count`, and `md5sum` return non-nullable result.
Therefore, I rename `nonNullResultWithNullParamFunctions` to `nonNullResultWithNullParamNullFunctions`, remove `hll_hash`, `bitmap_count`, and `md5sum` from `nonNullResultWithNullParamNullFunctions`, and modify `isNonNullResultWithNullParamFunctions()` as follows:
```C++
    public boolean isNonNullResultWithNullParamFunctions(String funcName) {
        return nonNullResultWithNullParamNullFunctions.contains(funcName)
                || alwaysReturnNonNullableFunctions.contains(funcName);
    }
```